esp_schedule/CMakeLists.txt

```
@@ -0,0 +1,10 @@
+set(component_srcs "src/esp_schedule.c"
+                   "src/esp_schedule_nvs.c")
+
+idf_component_register(SRCS "${component_srcs}"
+                       INCLUDE_DIRS "include"
+                       PRIV_INCLUDE_DIRS "src"
+                       PRIV_REQUIRES "rmaker_common"
+                       REQUIRES "nvs_flash" "esp_daylight")
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-function")
```

Recommend using 

```suggestion
target_compile_options(${COMPONENT_LIB} PRIVATE -Wno-unused-function)
```
instead of manipulating CMake variables